internal/character/silverwolf/eidolon.go

```diff
@@ -0,0 +1,106 @@
+package silverwolf
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	E2 key.Modifier = "silverwolf-e2"
+	E6 key.Modifier = "silverwolf-e6"
+)
+
+func init() {
+	// When an enemy enters battle, reduces their Effect RES by 20%.
+	modifier.Register(E2, modifier.Config{
+		Stacking:   modifier.ReplaceBySource,
+		StatusType: model.StatusType_STATUS_DEBUFF,
+		TickMoment: modifier.ModifierPhase1End,
+		Listeners: modifier.Listeners{
+			OnAdd: func(mod *modifier.ModifierInstance) {
+				mod.SetProperty(prop.EffectRES, -0.2)
+			},
+		},
+	})
+
+	// For every debuff the target enemy has, the DMG dealt by Silver Wolf increases by 20%, up to a limit of 100%.
+	modifier.Register(E6, modifier.Config{
+		TickMoment: modifier.ModifierPhase1End,
+		Stacking:   modifier.ReplaceBySource,
+		StatusType: model.StatusType_STATUS_DEBUFF,
+		Listeners: modifier.Listeners{
+			OnBeforeHitAll: func(mod *modifier.ModifierInstance, e event.HitStartEvent) {
+				debuffCount := mod.Engine().ModifierCount(e.Defender, model.StatusType_STATUS_DEBUFF)
+				if debuffCount > 5 {
+					debuffCount = 5
+				}
+				e.Hit.Attacker.AddProperty(prop.AllDamagePercent, 0.2*float64(debuffCount))
+			},
+		},
+	})
+}
+
+// After using her Ultimate to attack enemies, Silver Wolf regenerates 7
+// Energy for every debuff that the target enemy currently has. This effect
+// can be triggered up to 5 time(s) in each use of her Ultimate.
+func (c *char) e1(target key.TargetID) {
+	if c.info.Eidolon >= 1 {
+		debuffCount := c.engine.ModifierCount(target, model.StatusType_STATUS_DEBUFF)
+		if debuffCount > 5 {
+			debuffCount = 5
+		}
```

Comment:
nit: move the `debuffCount` check + limit to ult and pass it in as a param to both e1 and e4 (can wrap the count in `eidolon >= 1` check so count only being done on E1+)

Reply:
This was done intentionally as I'm not sure if e4 can proc new debuffs before e1 does its own debuff check. If you can guarantee that it won't add more debuffs, I can go through with this change.